### PR TITLE
Bar Chart Example Page

### PR DESCRIPTION
**Pull request by SybelBlue**
I combined histogram.tsx and scatterplot.tsx along with the [observablehq example](https://observablehq.com/@observablehq/plot-bar?collection=@observablehq/plot) to create a simple, alphabetic bar chart example page!

### CODE REVIEW

File: data/alphabet.json [hunk 1]
```diff
@@ -0,0 +1 @@
+[{"letter":"E","frequency":0.12702},{"letter":"T","frequency":0.09056},{"letter":"A","frequency":0.08167},{"letter":"O","frequency":0.07507},{"letter":"I","frequency":0.06966},{"letter":"N","frequency":0.06749},{"letter":"S","frequency":0.06327},{"letter":"H","frequency":0.06094},{"letter":"R","frequency":0.05987},{"letter":"D","frequency":0.04253},{"letter":"L","frequency":0.04025},{"letter":"C","frequency":0.02782},{"letter":"U","frequency":0.02758},{"letter":"M","frequency":0.02406},{"letter":"W","frequency":0.0236},{"letter":"F","frequency":0.02288},{"letter":"G","frequency":0.02015},{"letter":"Y","frequency":0.01974},{"letter":"P","frequency":0.01929},{"letter":"B","frequency":0.01492},{"letter":"V","frequency":0.00978},{"letter":"K","frequency":0.00772},{"letter":"J","frequency":0.00153},{"letter":"X","frequency":0.0015},{"letter":"Q","frequency":0.00095},{"letter":"Z","frequency":0.00074}]
```
Comment (parkerziegler) [inline data/alphabet.json]: Small nit on this JSON file and `crimea.json` — let's get these auto-formatted so they're easy to read and not all on one line!

In `package.json`, let's change the `format` script (under the `"scripts"` key) from:

```json
"format": "prettier --write src/**/*.ts"
```

to:

```json
"format": "prettier --write **/*.{ts,tsx} **/*.json"
```

and then just run `yarn format` and commit the result! I'll add a pre-commit hook to do this // do it in CI in the future.

File: tsconfig.json [hunk 2]
```diff
@@ -25,7 +25,7 @@
   },
   "include": [
     "src/**/*"
-  ],
+, "pages/helpers.ts"  ],
```
Comment (parkerziegler) [inline tsconfig.json]: So funky thing about Next.js is that treats everything in the `pages` directory as a route. So you can go to `localhost:3000/helpers` and it'll actually attempt to render a page from that 😱
 
image

This is all to say, we should put this in a separate folder. I'm thinking a top-level `helpers` right at the root should be fine.

File: pages/helpers.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import * as fs from 'fs';
+import * as path from 'path';
+
+export function readData<T extends object>(json_name: string): T[] {
```
Comment (parkerziegler) [inline pages/helpers.ts]: Small nit that I'll try to move into a lint rule — JS is typically camelCase land 🐫!

```suggestion
export function readData<T extends object>(jsonName: string): T[] {
```

File: pages/helpers.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+  const json = fs
+    .readFileSync(path.join(process.cwd(), `data/${json_name}.json`))
+    .toString();
+  return JSON.parse(json) as T[];
```
Comment (parkerziegler) [inline pages/helpers.ts]: Remind me to tell you about `io-ts` and runtime type decoding of JSON at some point (more for funzies, it'd be overkill on this project 😂).

**Review by parkerziegler — COMMENTED** [review]
Really nice work @SybelBlue! All of the feedback I have is related to small nits, formatting, Next.js weirdness, but the examples look ace! Let's make those few small changes and then LGTM 💯

File: data/alphabet.json (hunk 1 above)
Comment (SybelBlue) [reply data/alphabet.json]: Done and done!

File: tsconfig.json (hunk 2 above)
Comment (SybelBlue) [reply tsconfig.json]: Moved, imports refactored, and pushed!

File: pages/helpers.ts (hunk 3 above)
Comment (SybelBlue) [reply pages/helpers.ts]: Changed!

File: pages/helpers.ts (hunk 4 above)
Comment (SybelBlue) [reply pages/helpers.ts]: Ooh that sounds super cool. I will be sure to!

**Comment by SybelBlue** [comment]
Because I didn't know that commits are consistently added to the branch, this PR now contains all three new examples, all of the data for each, and some supporting helpers/config tweaks.

File: pages/examples/strip-plot.tsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+  age: AgeRange,
+  population: number,
+}
+
+interface Props {
+  states: StateDistribution[],
+  stateAges: StateAge[],
+}
+
+const Histogram: React.FC<Props> = ({ stateAges }) => {
```
Comment (parkerziegler) [inline pages/examples/strip-plot.tsx]: Let's make sure we change this name to reflect the right example!

```suggestion
const StripPlot: React.FC<Props> = ({ stateAges }) => {
```

File: pages/examples/bar.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+interface Letter {
+  letter: string,
+  frequency: number,
+}
+
+interface Props {
+  data: Letter[]
+}
+
+const Histogram: React.FC<Props> = ({ data }) => {
```
Comment (parkerziegler) [inline pages/examples/bar.tsx]: Let's change this component's name as well:

```suggestion
const BarChart: React.FC<Props> = ({ data }) => {
```

File: pages/examples/stacked-bar.tsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  date: string,
+  cause: string,
+  deaths: number,
+}
+
+interface Props {
+  data: DeathRecord[]
+}
+
+const Histogram: React.FC<Props> = ({ data }) => {
```
Comment (parkerziegler) [inline pages/examples/stacked-bar.tsx]: And here!
```suggestion
const StackedBarChart: React.FC<Props> = ({ data }) => {
```

**Review by parkerziegler — APPROVED** [review]

**Pull request merged**